csharp/src/Microsoft.ML.OnnxRuntime/OrtValue.shared.cs

```diff
@@ -339,34 +339,30 @@ private static OrtValue CreateStringTensor(Tensor<string> tensor)
                                 out valueHandle
                                 ));
 
-            var ortValue = new OrtValue(valueHandle);
+            OrtValue ortValue = null;
+            MarshaledStringArray marshaledStrings = default;
+
             try
             {
+                marshaledStrings = new MarshaledStringArray(tensor);
+                ortValue = new OrtValue(valueHandle);
```

Comment:
>ortValue = new OrtValue(valueHandle); [](http://example.com/codeflow?start=16&length=37)

Always been curious as why this new is moved under try and then we have to check for null in the finally/catch?

Reply:
It avoids an extra allocation in the case of failure or if the constructor itself fails. It's not strictly necessary, just generally considered "good practice" to handle it that way.